src/ayb_db/models.rs

```diff
@@ -2,7 +2,37 @@ use clap::ValueEnum;
 use serde::{Deserialize, Serialize};
 use serde_repr::{Deserialize_repr, Serialize_repr};
 use sqlx::FromRow;
-use std::fmt;
+use std::str::FromStr;
+
+macro_rules! try_from_i16 {
+    ($struct:ident, { $($left:literal => $right:expr),+ }) => {
+        impl TryFrom<i16> for $struct {
+            type Error = ();
+
+            fn try_from(value: i16) -> Result<Self, Self::Error> {
+                match value {
+                    $($left => Ok($right),)*
+                    _ => Err(()),
+                }
+            }
+        }
+    };
+}
+
+macro_rules! from_str {
```

Comment:
For naming consistency with the macro above

```suggestion
macro_rules! try_from_str {
```

Reply:
I named the macros according to the trait they implement.
`TryFrom<i16>` &rarr; `try_from_i16`
`FromStr` &rarr; `from_str`

I don't think we should change that, I'll mark the conversation as resolved if you agree

Reply:
Sounds great, thank you! :)